src/pages/home/ReportScreen.js

```diff
@@ -123,6 +127,64 @@ class ReportScreen extends React.Component {
         Navigation.setIsReportScreenIsReady();
     }
 
+    shouldComponentUpdate(nextProps) {
+        if (!_.isEqual(nextProps.reportActions, this.props.reportActions)) {
+            this.sortedAndFilteredReportActions = this.getSortedReportActionsForDisplay(nextProps.reportActions);
+
+            return true;
+        }
```

Comment:
@gedu I am not very happy with moving all of this logic up which mens amore components will have to re-render because of this. Could we keep it in the `ReportActionsView` and move the `getSortedReportActionsForDisplay` method to `ReportActionsUtils` instead since it is itself utils method.

Then we could get the sorted report actions in the ReportScreen, pass it down to the Report screen components and we can call the `ReportActionsUtils.getSortedReportActionsForDisplay` from the `ReportActionsView` if needed.

Is there anything I am missing?

Reply:
@mountiny We needed to move up so the `ReportActionsView` and `ReportFooter` receives the same sorted list. 

I could move `getSortedReportActionsForDisplay` into a utils file, but we should keep the sorting at the top. 

About the rerender, it will be the same as before, because when `ReportScreen` needs to rerender when `reportActions` changes, was rerendering both components `ReportActionsView` and `ReportFooter`..., the first one to sort the actions again, and the Footer because of this arrow up feature among others props.

Reply:
Thanks for context.